pkg/loki/loki.go

```diff
@@ -111,6 +114,8 @@ func (c *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&c.UseBufferedLogger, "log.use-buffered", true, "Uses a line-buffered logger to improve performance.")
 	f.BoolVar(&c.UseSyncLogger, "log.use-sync", true, "Forces all lines logged to hold a mutex to serialize writes.")
 
+	f.BoolVar(&c.LegacyReadTarget, "legacy-read-mode", false, "Set to true to enable the legacy read mode. This will be removed in Loki 3.0.")
```

Comment:
I wasn't sure which direction to go here. On by default is less disruptive, but might create more pain when we want to remove it in 3.0? Or maybe we flip it in 3.0, and remove it later?

Reply:
Yes, I vote for having it as truthy by default, otherwise it will break setups without users expecting it. 

Regarding flipping it in 3.0 and removing it later or removing it entirely in 3.0, both are ok in my opinion.

Reply:
Cool, agreed :+1: flipped!